data-structures.md

```diff
@@ -221,7 +221,107 @@ Bitfields are a set of bits encoded using a custom run length encoding: rle+.  r
 
 #### `SectorSet`
 
-TODO
+The `SectorSet` is an integer set implemented with a simple array mapped tree.
+Integer indexes range from 0 to infinity (TODO practical bounds / bounds implied by encoding?).
+The two basic structures in the `SectorSet` are the `Node` and
+the `Pointer`. There is also a `RootNode`.   The `Node` is as follows:
+```go
+type Node struct {
+     Pointers [S]*Pointer
+}
+```
+where `S` is the fixed node width.  `S` is a constant set to `256`.  The `Node`
+is serialized as a cbor array (major type 4) of Pointers.
+
+
+[ipld dag-cbor Cids](https://github.com/ipld/specs/blob/master/Codecs/DAG-CBOR.md#link-format).
+
+The `Pointer` is as follows:
+```go
+type Pointer struct {
+     Value SectorMeta
+     Link Cid
+     
+}
+```
+The `Pointer` is serialized as a cbor object (major type 5). The  `Value`
+is serialized { in some way } with `v` as its object key.  The `SectorMeta`
+type carries all needed sector metadata (TODO, define a type for commD commR
```

Comment:
Is this (commR, commD)?

Reply:
cc @laser

Reply:
`SectorMeta` will need:

- CommR so that we can generate PoSts (the `generate_post` routines takes as input all CommRs in the miner's proving set).
- CommD for piece inclusion proof verification.